<a href="https://colab.research.google.com/github/jsvan/MaskTests4LanguageModels/blob/main/AllenAI's_Git_code_version_of_my_Extension_of_Summerwinograndeattempt_to_more_models.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In this extension I will try out more models, first tuning generic language models on the debiased dataset. 

In [ ]:
%%capture
!pip install transformers
!pip install datasets 

In [2]:
!git clone https://github.com/allenai/winogrande.git

Cloning into 'winogrande'...
remote: Enumerating objects: 41, done.
remote: Counting objects: 100% (41/41), done.
remote: Compressing objects: 100% (29/29), done.
remote: Total 41 (delta 16), reused 33 (delta 11), pack-reused 0
Unpacking objects: 100% (41/41), done.


In [3]:
!chmod 755 -R winogrande/sample_training.sh
%cd winogrande
!ls

/content/winogrande
Dockerfile		       sample_prediction.sh
download_winogrande.sh	       sample_prediction_superglue-wsc.sh
LICENSE			       sample_training.sh
predict_winogrande_on_bkr.sh   sample_training_superglue-wsc.sh
predict_winogrande_on_bkr.yml  scripts
README.md		       train_winogrande_on_bkr.sh
requirements.txt	       train_winogrande_on_bkr.yml


In [4]:
!ls
with open("./scripts/run_experiment.py", "r") as F:
  f = F.readlines()

f[53] = """\nimport sys, os
sys.path.append("/content/winogrande/")
sys.path.append("/content/winogrande/scripts/")\n\n"""

with open("./scripts/run_experiment.py", "w") as F:
  F.write(''.join(f))

Dockerfile		       sample_prediction.sh
download_winogrande.sh	       sample_prediction_superglue-wsc.sh
LICENSE			       sample_training.sh
predict_winogrande_on_bkr.sh   sample_training_superglue-wsc.sh
predict_winogrande_on_bkr.yml  scripts
README.md		       train_winogrande_on_bkr.sh
requirements.txt	       train_winogrande_on_bkr.yml


In [5]:
!pip install -r requirements.txt

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 158 kB 4.1 MB/s 
     |████████████████████████████████| 125 kB 34.4 MB/s 
     |████████████████████████████████| 132 kB 64.4 MB/s 
     |████████████████████████████████| 1.2 MB 64.4 MB/s 
     |████████████████████████████████| 9.0 MB 59.4 MB/s 
     |████████████████████████████████| 79 kB 9.4 MB/s 
     |████████████████████████████████| 127 kB 74.8 MB/s 
  Attempting uninstall: urllib3
    Found existing installation: urllib3 1.24.3
    Uninstalling urllib3-1.24.3:
      Successfully uninstalled urllib3-1.24.3
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
datascience 0.10.6 requires folium==0.2.1, but you have folium 0.8.3 which is incompatible.


/content/winogrande


In [6]:
!ls

Dockerfile		       sample_prediction.sh
download_winogrande.sh	       sample_prediction_superglue-wsc.sh
LICENSE			       sample_training.sh
predict_winogrande_on_bkr.sh   sample_training_superglue-wsc.sh
predict_winogrande_on_bkr.yml  scripts
README.md		       train_winogrande_on_bkr.sh
requirements.txt	       train_winogrande_on_bkr.yml


In [8]:
!chmod 755 -R ./download_winogrande.sh
!./download_winogrande.sh

--2022-07-07 16:05:45--  https://storage.googleapis.com/ai2-mosaic/public/winogrande/winogrande_1.1.zip
Resolving storage.googleapis.com (storage.googleapis.com)... 74.125.23.128, 74.125.203.128, 74.125.204.128, ...
Connecting to storage.googleapis.com (storage.googleapis.com)|74.125.23.128|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3395492 (3.2M) [application/zip]
Saving to: ‘winogrande_1.1.zip’

winogrande_1.1.zip  100%[===================>]   3.24M  --.-KB/s    in 0.02s   

2022-07-07 16:05:46 (147 MB/s) - ‘winogrande_1.1.zip’ saved [3395492/3395492]

Archive:  winogrande_1.1.zip
   creating: winogrande_1.1/
  inflating: winogrande_1.1/train_debiased-labels.lst  
  inflating: winogrande_1.1/train_m.jsonl  
  inflating: winogrande_1.1/train_l-labels.lst  
  inflating: winogrande_1.1/test.jsonl  
  inflating: winogrande_1.1/train_l.jsonl  
  inflating: winogrande_1.1/train_xs-labels.lst  
  inflating: winogrande_1.1/sample-submission-labels.lst  
  infla

In [ ]:
!./sample_training.sh

07/07/2022 16:05:59 - WARNING - __main__ -   Process rank: -1, device: cuda, n_gpu: 1, distributed training: False, 16-bits training: False
07/07/2022 16:06:00 - INFO - pytorch_transformers.file_utils -   https://s3.amazonaws.com/models.huggingface.co/bert/roberta-large-config.json not found in cache, downloading to /tmp/tmpdbf_784j
100% 482/482 [00:00<00:00, 475012.81B/s]
07/07/2022 16:06:00 - INFO - pytorch_transformers.file_utils -   copying /tmp/tmpdbf_784j to cache at /root/.cache/torch/pytorch_transformers/c22e0b5bbb7c0cb93a87a2ae01263ae715b4c18d692b1740ce72cacaa99ad184.2d28da311092e99a05f9ee17520204614d60b0bfdb32f8a75644df7737b6a748
07/07/2022 16:06:00 - INFO - pytorch_transformers.file_utils -   creating metadata file for /root/.cache/torch/pytorch_transformers/c22e0b5bbb7c0cb93a87a2ae01263ae715b4c18d692b1740ce72cacaa99ad184.2d28da311092e99a05f9ee17520204614d60b0bfdb32f8a75644df7737b6a748
07/07/2022 16:06:00 - INFO - pytorch_transformers.file_utils -   removing temp file /tmp/t

In [ ]:
# Textizer
"""
    Each sentence was split on "_" placeholder symbol.
    Each option was concatenated with the second part of the split, thus transforming each example into two text segment pairs.
    Text segment pairs corresponding to correct and incorrect options were marked with True and False labels accordingly.
    Text segment pairs were shuffled thereafter.

"""

from datasets import Dataset

def prepare_data(dataset):

  # internal function
  def prep_ds(dataset):
    sentences, answers, o1, o2 = [], [], [], []
    for p in dataset:
      s1 = p['option1'].join(p['sentence'].split('_'))
      s2 = p['option2'].join(p['sentence'].split('_'))
      a1 = int(p['answer'] == '1')
      a2 = int(p['answer'] == '2')

      
      sentences.append(s1)
      answers.append(a1)
      sentences.append(s2)
      answers.append(a2)
      o1.append(p['option1'])
      o2.append(p['option2'])
      o1.append(p['option1'])
      o2.append(p['option2'])

    return {'sentence':sentences, 'labels':answers, 'option1':o1, 'option2':o2}
    # end internal function

  train = prep_ds(dataset["train"])
  test = prep_ds(dataset["validation"])
  trainds = Dataset.from_dict( train ).shuffle()
  testds = Dataset.from_dict( test ).shuffle()

  return {"train":trainds, "test":testds, "name":"Standard Dataset"}


In [ ]:

def mask_datasets(dataset):

  # internal method BEGIN
  def mask_copy(dataset):
    sentences = []
    toprint = 1

    for p in dataset:
      if toprint > 0:
        print(f"[{p['option1']}], [{p['option2']}], [{p['sentence']}]")

      sentences.append(p['sentence'].replace(p['option1'], 'option1').replace(p['option2'], 'option2'))
      
      if toprint > 0:
        print(sentences[-1])
        toprint -= 1

    build = {'sentence':sentences, 'labels':dataset['labels'], 'option1':dataset['option1'], 'option2':dataset['option2']}
    return Dataset.from_dict(build) # DON'T SHUFFLE
  #END
  
  return {"train":mask_copy(dataset['train']), "test":mask_copy(dataset['test']), "name":"Masked Dataset"}



In [ ]:
from transformers import (AutoTokenizer, 
                                  AutoModelForSequenceClassification, 
                                  #T5ForConditionalGeneration, 
                                  #T5Tokenizer,
                                  get_linear_schedule_with_warmup,
                                  AdamW)
import torch
from pprint import pprint
from tqdm import tqdm

""" 
from transformers import AutoTokenizer, AutoModelForMaskedLM

tokenizer = AutoTokenizer.from_pretrained("roberta-large")

model = AutoModelForMaskedLM.from_pretrained("roberta-large")
"""


def test_datasets(tokenizer, std_datasets, masked_datasets, model=False,):

  print("\nTESTING TESTING  1 2 3 ...")
  delete = False
  if not model:
    #try:
    model = AutoModelForSequenceClassification.from_pretrained(model)
    #except:
    #  model = T5ForConditionalGeneration.from_pretrained(model)
    
    delete = True

  elif isinstance(model, str):
    delete = True
    with torch.no_grad():
      torch.cuda.empty_cache()

    model = torch.load(model) # open(model, "rb"))

    with torch.no_grad():
      torch.cuda.empty_cache()
  try:
    for ds in (std_datasets, masked_datasets):
      print("\nPERFORMING", ds["name"])
      
      encoded_test = ds['test'].map(lambda examples: tokenizer(examples['sentence'], padding='max_length'), batched=True)

      encoded_test.set_format(type='torch', columns=['input_ids', 'attention_mask', 'labels'])
      dataloader_test = torch.utils.data.DataLoader(encoded_test, batch_size=32)

      device = 'cuda' if torch.cuda.is_available() else 'cpu' 
      model.to(device)

      correct = 0
      total = 0
      for i, batch in enumerate(tqdm(dataloader_test)):
        batch = {k: v.to(device) for k, v in batch.items()}

        outputs = model(**batch)
        #for oo, lab in zip(outputs.logits, batch['labels'] ):
        #  print(oo.argmax().item(), lab.items())
        correct += sum((int(x.argmax().item() == y.item()) for x, y in zip(outputs.logits, batch['labels'])))
        total += len(batch['labels'])
        
        
      print("\nSCORE", correct / total, '/ 1.00')

    if delete:
      del model
      print("Deleted model")

  except Exception as e:
    del model
    print(e, e.__str__)
    print("Deleted model")




In [ ]:

def train_model(trainingset, compareset, tokenizer, model=False, copy=False, epochs=1):
  print('\n\nTRAIN:\n', '\n'.join(trainingset['train']['sentence'][0:10]))
  print('\n\nLABELS:\n', '\n'.join([str(x) for x in trainingset['train']['labels'][0:10]]))

  print('\n\nTEST:\n', '\n'.join(trainingset['test']['sentence'][0:10]))
  print('\n\nLABELS:\n', '\n'.join([str(x) for x in trainingset['test']['labels'][0:10]]))


  if copy: # ie, if copy and model
    print('DUPLICATING MODEL')

    with torch.no_grad():
      torch.cuda.empty_cache()
    model = torch.load(model) # open(model, "rb"))
    
  elif model and isinstance(model, str): # ie if str(model) AND NOT copy:
    print("DOWNLOADING MODEL")
    # download new model of name
    #try:
    model = AutoModelForSequenceClassification.from_pretrained(model)
    #except ValueError:
    #  model = T5ForConditionalGeneration.from_pretrained(model)

  try:
    encoded_train = trainingset['train'].map(lambda examples: tokenizer(examples['sentence'], padding='max_length'), batched=True) # , return_tensors='pt'
    encoded_train.set_format(type='torch', columns=['input_ids', 'attention_mask', 'labels'])
    dataloader_train = torch.utils.data.DataLoader(encoded_train, batch_size=32)

    device = 'cuda' if torch.cuda.is_available() else 'cpu' 
    model.train().to(device)
    optimizer = AdamW(params=model.parameters(), lr = 1e-5)
    scheduler = get_linear_schedule_with_warmup(optimizer = optimizer, 
                                     num_warmup_steps = 0.1 * len(dataloader_train) * epochs, 
                                     num_training_steps = len(dataloader_train) * epochs)
    
    for epoch in range(epochs):
      print("EPOCH", epoch+1, '/', epochs)
      correct = 0
      total = 0
      for i, batch in enumerate(tqdm(dataloader_train)):
        batch = {k: v.to(device) for k, v in batch.items()}

        outputs = model(**batch)
        loss = outputs[0]
        loss.backward()
        optimizer.step()
        scheduler.step()
        optimizer.zero_grad()

        if i < 1:
          for oo, lab in zip(outputs.logits, batch['labels'] ):
            print('(', oo.argmax().item(), ',', lab.item(), ')', end=',  ')
          print()
        correct += sum((int(x.argmax().item() == y.item()) for x, y in zip(outputs.logits, batch['labels'])))
        total += len(batch['labels'])
        
      print("Score", correct / total, '/ 1.00')
      test_datasets(tokenizer, trainingset, compareset, model)
      model.train().to(device)
  except Exception as e:
    del model
    print(e, e.__str__)
    print("Error on line {}".format(sys.exc_info()[-1].tb_lineno))
    print("deleted model")
  if copy:
    del model
  else:
    return model




In [ ]:
import re

def mask_copy_1(dataset, unk):
  sentences = []
  toprint = 1

  for p in dataset:
    # This is annoying because option1 can be substrings of other words, usually option2. They can also have uppercase letters. 
    # If one is a substring, then I will cover up the larger word with a temporary mask to not confuse anything else.
    sentence = p['sentence']
    option1, option2 = re.compile(p['option1'], re.IGNORECASE), re.compile(p['option2'], re.IGNORECASE)
    maskedoption1, maskedoption2 = "OPTION_ONE", "OPTION_TWO"
    first_search, second_search, first_mask, second_mask = None, None, None, None

    # "table" in "tablecloth" --> cover bigger one
    if len(p['option1']) > len(p['option2']):
      # cover option1 first
      first_search, second_search = option1, option2
      first_mask, second_mask = maskedoption1, maskedoption2
    else:
      first_search, second_search = option2, option1
      first_mask, second_mask = maskedoption2, maskedoption1
    
    sentence = first_search.sub(first_mask, sentence) #Mask the longer word with OPTION_ mask
    sentence = second_search.sub(second_mask, sentence) #then the shorter one

    # IT gets kinda confusing which word now to <IGNORE> because it's language and some words appear more than two, more than three, times. 
    # I think the smartest approach is to assume the final word used is the question word and mask that one. 

    # Find final word used
    # maskedoption1 is the final word
    if sentence.rfind(maskedoption1) > sentence.rfind(maskedoption2):
      # IGNORE final word
      # Convert other word back to original
      sentence = sentence.replace(maskedoption1, unk)
      sentence = sentence.replace(maskedoption2, p['option2'])
    else:
      sentence = sentence.replace(maskedoption2, unk)
      sentence = sentence.replace(maskedoption1, p['option2'])


    if toprint > 0:
      print(f"[{p['option1']}], [{p['option2']}], [{p['sentence']}]")

    sentences.append(sentence)
    
    if toprint > 0:
      print(sentences[-1])
      toprint -= 1

  build = {'sentence':sentences, 'labels':dataset['labels'], 'option1':dataset['option1'], 'option2':dataset['option2']}
  return Dataset.from_dict(build) # DON'T SHUFFLE





def mask_datasets_1(dataset, tokenizer):
  unk = tokenizer.unk_token
  return {"train":mask_copy_1(dataset['train'], unk), "test":mask_copy_1(dataset['test'], unk), "name":"Double <Unk> Masked Dataset"}

In [ ]:
import re


# This masked <unk> tokens on the option which is NOT involved in the question. 
# If the option involved in the question is wrong, it will need to identify the <unk> token as having importance.
def mask_copy_2(dataset, unk):
  sentences = []
  toprint = 1

  for p in dataset:
    # This is annoying because option1 can be substrings of other words, usually option2. They can also have uppercase letters. 
    # If one is a substring, then I will cover up the larger word with a temporary mask to not confuse anything else.
    sentence = p['sentence']
    option1, option2 = re.compile(p['option1'], re.IGNORECASE), re.compile(p['option2'], re.IGNORECASE)
    maskedoption1, maskedoption2 = "OPTION_ONE", "OPTION_TWO"
    first_search, second_search, first_mask, second_mask = None, None, None, None

    # "table" in "tablecloth" --> cover bigger one
    if len(p['option1']) > len(p['option2']):
      # cover option1 first
      first_search, second_search = option1, option2
      first_mask, second_mask = maskedoption1, maskedoption2
    else:
      first_search, second_search = option2, option1
      first_mask, second_mask = maskedoption2, maskedoption1
    
    sentence = first_search.sub(first_mask, sentence) #Mask the longer word with OPTION_ mask
    sentence = second_search.sub(second_mask, sentence) #then the shorter one

    # IT gets kinda confusing which word now to <IGNORE> because it's language and some words appear more than two, more than three, times. 
    # I think the smartest approach is to assume the final word used is the question word and mask that one. 

    # Find final word used
    # maskedoption1 is NOT the final word
    if sentence.rfind(maskedoption1) < sentence.rfind(maskedoption2):
      # IGNORE final word
      # Convert other word back to original
      sentence = sentence.replace(maskedoption1, unk)
      sentence = sentence.replace(maskedoption2, p['option2'])
    else:
      sentence = sentence.replace(maskedoption2, unk)
      sentence = sentence.replace(maskedoption1, p['option2'])


    if toprint > 0:
      print(f"[{p['option1']}], [{p['option2']}], [{p['sentence']}]")

    sentences.append(sentence)
    
    if toprint > 0:
      print(sentences[-1])
      toprint -= 1

  build = {'sentence':sentences, 'labels':dataset['labels'], 'option1':dataset['option1'], 'option2':dataset['option2']}
  return Dataset.from_dict(build) # DON'T SHUFFLE





def mask_datasets_2(dataset, tokenizer):
  unk = tokenizer.unk_token
  return {"train":mask_copy_2(dataset['train'], unk), "test":mask_copy_2(dataset['test'], unk), "name":"Single <Unk> Masked Dataset"}

In [ ]:
stupid_tries = [
  ("dog", "doggy"),
  ("red", "blue"),
  ("flavor", "flavour"),
  ('A', 'B'),
  ('X', 'Y'),
  ('1', '2'),
  ('first', 'second'),
  ('alpha', 'beta'),
  ('#', '@'),
  ('primero', 'secundo'), # yes its true, I dont speak spanish. Having it spelled wrong just makes the model's task that much more difficult ;)
  ('Alice', 'Bob'),
  ('_', '__'),
  ('mask1', 'mask2'),
  ('thing1', 'thing2'),
  ('mask_a', 'mask_b'),
  ("thing_a", "thing_b")
]


In [ ]:
import re


# This masked <unk> tokens on the option which is NOT involved in the question. 
# If the option involved in the question is wrong, it will need to identify the <unk> token as having importance.
def stupid_masking(dataset, mask1, mask2):
  sentences = []
  toprint = 1

  for p in dataset:
    # This is annoying because option1 can be substrings of other words, usually option2. They can also have uppercase letters. 
    # If one is a substring, then I will cover up the larger word with a temporary mask to not confuse anything else.
    sentence = p['sentence']
    option1, option2 = re.compile(p['option1'], re.IGNORECASE), re.compile(p['option2'], re.IGNORECASE)
    maskedoption1, maskedoption2 = "OPTION_ONE", "OPTION_TWO"
    first_search, second_search, first_mask, second_mask = None, None, None, None

    # "table" in "tablecloth" --> cover bigger one
    if len(p['option1']) > len(p['option2']):
      # cover option1 first
      first_search, second_search = option1, option2
      first_mask, second_mask = maskedoption1, maskedoption2
    else:
      first_search, second_search = option2, option1
      first_mask, second_mask = maskedoption2, maskedoption1
    
    sentence = first_search.sub(first_mask, sentence) #Mask the longer word with OPTION_ mask
    sentence = second_search.sub(second_mask, sentence) #then the shorter one

    sentence = sentence.replace(maskedoption1, mask1)
    sentence = sentence.replace(maskedoption2, mask2)


    if toprint > 0:
      print(f"[{p['option1']}], [{p['option2']}], [{p['sentence']}]")

    sentences.append(sentence)
    
    if toprint > 0:
      print(sentences[-1])
      toprint -= 1

  build = {'sentence':sentences, 'labels':dataset['labels'], 'option1':dataset['option1'], 'option2':dataset['option2']}
  return Dataset.from_dict(build) # DON'T SHUFFLE





def stupid_datasets(dataset, tokenizer, masks):
  mask1, mask2 = masks
  return {"train":stupid_masking(dataset['train'], mask1, mask2), "test":stupid_masking(dataset['test'], mask1, mask2), "name":f"({mask1}, {mask2}) Masked Dataset"}

In [ ]:
%%capture
from datasets import load_dataset, concatenate_datasets

# You can switch between these two datasets
dataset = load_dataset("winogrande", 'winogrande_debiased')
#dataset = load_dataset("winogrande", 'winogrande_l')

# dataset is dict with keys ['train', 'test', 'validation']
# Each with an enumerable of 
"""
{'answer': '2',
 'option1': 'Kyle',
 'option2': 'Logan',
 'sentence': "Kyle doesn't wear leg warmers to bed, while Logan almost always does. _ is more likely to live in a colder climate."}

"""
# Use Validation instead of Test because Test lacks labels.


#dicts of {'train':_, 'test':_}
std_datasets = prepare_data(dataset)

masked_datasets = mask_datasets(std_datasets)

In [ ]:
import os
import sys
# This model will become the base model for everything we build upon, 
# so we will save it to disk to load it fresh for each experiment. 
model_list = [#"DeepPavlov/roberta-large-winogrande", 
              #"t5-small",
              #"t5-large",
              "roberta-large",
              "distilbert-base-uncased", 
              "albert-base-v2", 
              'bert-base-cased', 
              'xlnet-base-cased', 
            ]

for model_name_orig in model_list:
  model_name = ''.join([c for c in model_name_orig if c.isalpha()])
  try:
    print("NOW DOING", model_name_orig)
    # download model and save to memory. Models must be removed from cache to prevent gpu mem errors.
    # fine tune the model to the debiased dataset
    tokenizer = AutoTokenizer.from_pretrained(model_name_orig, model_max_length=64)
    print("BEGIN OG TRAIN")
    debiased_tuned_model = train_model(trainingset=std_datasets, compareset=masked_datasets, 
                                       tokenizer=tokenizer, model=model_name_orig, copy=False, 
                                       epochs=5)



    print("SAVED")
    model_name = ''.join([c for c in model_name if c.isalpha()])
    torch.save(debiased_tuned_model, f=model_name)
    debiased_tuned_model = None


    # Run individual tests 
    print("STARTING FIRST UNK")
    unk_datasets = mask_datasets_1(std_datasets, tokenizer)
    test_datasets(tokenizer, std_datasets, unk_datasets, debiased_tuned_model)
    train_model(unk_datasets, std_datasets, tokenizer, model=model_name, copy = True)
    print("ENDING FIRST UNK & STARTING SECOND")

    unk_dataset_2 = mask_datasets_2(std_datasets, tokenizer)
    test_datasets(tokenizer, std_datasets, unk_dataset_2, debiased_tuned_model)
    train_model(unk_dataset_2, std_datasets, tokenizer, model=model_name, copy = True)

    print("ENDING UNK TESTS AND STARTING STUPID MASKS")

    for masks in stupid_tries[-2:]:
      stupid_ds = stupid_datasets(std_datasets, tokenizer, masks)
      print("TESTING:")
      test_datasets(tokenizer, std_datasets, stupid_ds, model=model_name)
      print("POST TRAINING:")
      train_model(stupid_ds, std_datasets, tokenizer, model=model_name, copy = True)

  finally:
    print("REMOVING MODEL")
    try:
      del tokenizer
      os.remove(model_name)
    except:
      print("finished")


  break

NOW DOING roberta-large


Downloading:   0%|          | 0.00/482 [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/878k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/446k [00:00<?, ?B/s]

Downloading:   0%|          | 0.00/1.29M [00:00<?, ?B/s]

BEGIN OG TRAIN


TRAIN:
 James was waiting for the alarm to go off so that he could listen to the radio. The radio is too loud.
Hunter had more aluminum foil in the house compared to Kevin, so Hunter let their leftovers be open.
More bottles could not be kept in the chests because the chests are not that small.
The eye doctor prescribed reading glasses to Samantha but not Emily because Samantha is keen-sighted.
Sarah had worse symptoms than Tanya, so Sarah was seen by the emergency room doctor second.
When Amy discovered the maxed out credit card, Carrie got embarrassed. So Carrie worked hard to pay it off.
Mary had to climb the mountain to reach Cynthia, so Mary could finally shower them with their wisdom.
Jeffrey was sick with the stomach flu and not hungry, but Christopher was starving. Christopher declined food.
Allison tried to use an elastic band to put her hair in a ponytail, but the elastic band was too short.
Jennifer earned royalties from the music that Katrina created, becau

Downloading:   0%|          | 0.00/1.33G [00:00<?, ?B/s]

Some weights of the model checkpoint at roberta-large were not used when initializing RobertaForSequenceClassification: ['lm_head.layer_norm.bias', 'lm_head.bias', 'lm_head.layer_norm.weight', 'roberta.pooler.dense.bias', 'roberta.pooler.dense.weight', 'lm_head.dense.bias', 'lm_head.decoder.weight', 'lm_head.dense.weight']
- This IS expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing RobertaForSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of RobertaForSequenceClassification were not initialized from the model checkpoint at roberta-large and are newly initialized: ['classifier.dense.bias', 'classifie

  0%|          | 0/19 [00:00<?, ?ba/s]

/usr/local/lib/python3.7/dist-packages/transformers/optimization.py:310: FutureWarning: This implementation of AdamW is deprecated and will be removed in a future version. Use the PyTorch implementation torch.optim.AdamW instead, or set `no_deprecation_warning=True` to disable this warning
  FutureWarning,


EPOCH 1 / 5


  0%|          | 1/578 [00:01<12:08,  1.26s/it]

( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 0 , 0 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 1 ),  ( 1 , 0 ),  ( 1 , 1 ),  ( 1 , 1 ),  ( 1 , 1 ),  


100%|██████████| 578/578 [10:59<00:00,  1.14s/it]

Score 0.5016760380622838 / 1.00

TESTING TESTING  1 2 3 ...

PERFORMING Standard Dataset


  0%|          | 0/3 [00:00<?, ?ba/s]

100%|██████████| 80/80 [00:31<00:00,  2.51it/s]


SCORE 0.5039463299131808 / 1.00

PERFORMING Masked Dataset


  0%|          | 0/3 [00:00<?, ?ba/s]

100%|██████████| 80/80 [00:31<00:00,  2.51it/s]



SCORE 0.4897395422257301 / 1.00
EPOCH 2 / 5


  0%|          | 1/578 [00:01<11:48,  1.23s/it]

( 1 , 0 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 0 , 1 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 1 , 0 ),  ( 0 , 1 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 1 , 1 ),  ( 0 , 1 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 1 , 1 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 0 , 1 ),  ( 0 , 0 ),  ( 0 , 1 ),  ( 0 , 0 ),  ( 0 , 0 ),  ( 1 , 1 ),  ( 0 , 0 ),  ( 1 , 1 ),  ( 0 , 1 ),  ( 0 , 0 ),  ( 0 , 1 ),  ( 1 , 1 ),  ( 0 , 1 ),  


100%|██████████| 578/578 [11:02<00:00,  1.15s/it]

Score 0.5025951557093425 / 1.00

TESTING TESTING  1 2 3 ...

PERFORMING Standard Dataset


  0%|          | 0/3 [00:00<?, ?ba/s]

100%|██████████| 80/80 [00:31<00:00,  2.51it/s]


SCORE 0.4980268350434096 / 1.00

PERFORMING Masked Dataset


  0%|          | 0/3 [00:00<?, ?ba/s]

100%|██████████| 80/80 [00:31<00:00,  2.51it/s]



SCORE 0.516179952644041 / 1.00
EPOCH 3 / 5


  0%|          | 0/578 [00:00<?, ?it/s]

Results:

| Masking Scheme | Accuracy (before training) | Accuracy (after training on silly set) |
|:-------:|:------------------:|:----------------:|
|Standard set| 75% | 75% |
|(Alice, Bob)| 73% | 73% |
|(primero, secundo)| 72% | 75% |
| (X, Y) | 72% | 74% |
| (A, B) | 72% | 73% |
| (1, 2) | 71% | 74% |
|(red, blue) | 71% | 73% |
|(#, @)| 71% | 73% |
|(alpha, beta)| 71% | 73% |
|(mask_a, mask_b) | 70% | 75% | 
|(thing_a, thing_b) | 70% | 73% |
|(first, second)| 69% | 73% |
|(\_, \_\_)| 67% | 73% |
|(dog, doggy)| 65% | 73% | 
|(flavor, flavour)| 53% | 74% |



| Masking Scheme | Accuracy (before training) | Accuracy (after training on \<unk\> masked set) |
|:-------:|:------------------:|:----------------:|
|\<unk\> as subj| 65% | 73% |
|\<unk\> as obj| 68% | 73% |


